### PR TITLE
Fix/improve clock in apic hal

### PR DESCRIPTION
**Pull request by tkreuzer**
This fixes a nasty timer assert. See CORE-16253

### CODE REVIEW

File: hal/halx86/apic/rtctimer.c [hunk 1]
```diff
@@ -113,15 +119,20 @@ HalpInitializeClock(VOID)
...
     __writeeflags(EFlags);
 
+    /* Calculate minumum and maximum increment */
+    HalpMinimumTimeIncrement = RtcClockRateToPreciseIncrement(RtcMinimumClockRate) / 1000;
+    HalpMaximumTimeIncrement = RtcClockRateToPreciseIncrement(RtcMaximumClockRate) / 1000;
+
     /* Notify the kernel about the maximum and minimum increment */
-    KeSetTimeIncrement(RtcClockRateToIncrement(RtcMaximumClockRate),
-                       RtcClockRateToIncrement(RtcMinimumClockRate));
+    KeSetTimeIncrement(HalpMaximumTimeIncrement, HalpMaximumTimeIncrement);
```
Comment (HBelusca) [inline hal/halx86/apic/rtctimer.c:132]:
```suggestion
    KeSetTimeIncrement(HalpMaximumTimeIncrement, HalpMinimumTimeIncrement);
```

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: hal/halx86/apic/rtctimer.c [hunk 2]
```diff
@@ -12,62 +12,68 @@
...
+    15       2    500            5,000,000,000
 
 */
 FORCEINLINE
 ULONG
-RtcClockRateToIncrement(UCHAR Rate)
+RtcClockRateToPreciseIncrement(UCHAR Rate)
 {
     /* Calculate frequency */
     ULONG Freqency = 32768 >> (Rate - 1);
```
Comment (HBelusca) [inline hal/halx86/apic/rtctimer.c:60]:
```suggestion
    ULONG Frequency = 32768 >> (Rate - 1);
```

File: hal/halx86/apic/rtctimer.c [hunk 3]
```diff
@@ -12,62 +12,68 @@
...
 
-    /* Calculate interval in 100ns interval: Interval = (1 / Frequency) * 10000000
-       This formula will round properly, instead of truncating. */
-    return (10000000 + (Freqency/2)) / Freqency;
+    /* Calculate interval in 0.1ns interval: Interval = (1 / Frequency) * 10,000,000,000 */
+    return 10000000000ULL / Freqency;
```
Comment (HBelusca) [inline hal/halx86/apic/rtctimer.c:63]:
```suggestion
    return 10000000000ULL / Frequency;
```

File: hal/halx86/apic/rtctimer.c [hunk 4]
```diff
@@ -12,62 +12,68 @@
...
+static ULONG HalpRunningFraction;
+static BOOLEAN HalpSetClockRate;
+static UCHAR HalpNextClockRate;
 
 /*!
     \brief Converts the CMOS RTC rate into the time increment in 100ns intervals.
 
-    Rate Freqency Interval (ms)  Result
-    -------------------------------------
+    Rate Freqency Interval (ms)  Precise increment (0.1ns)
```
Comment (HBelusca) [inline hal/halx86/apic/rtctimer.c:35]:
```suggestion
    Rate Frequency Interval (ms)  Precise increment (0.1ns)
```

File: hal/halx86/apic/tsc.c [hunk 5]
```diff
@@ -10,14 +10,14 @@
...
 #include <debug.h>
 
 LARGE_INTEGER HalpCpuClockFrequency = {{INITIAL_STALL_COUNT * 1000000}};
 
 UCHAR TscCalibrationPhase;
 ULONG64 TscCalibrationArray[NUM_SAMPLES];
-UCHAR HalpRtcClockVector = 0xD1;
 
 #define RTC_MODE 6 /* Mode 6 is 1024 Hz */
 #define SAMPLE_FREQENCY ((32768 << 1) >> RTC_MODE)
```
Comment (HBelusca) [inline hal/halx86/apic/tsc.c:23]:
```suggestion
#define SAMPLE_FREQUENCY ((32768 << 1) >> RTC_MODE)
```
+ typo fix in lines 47 and 75.

**Review by HBelusca — APPROVED** [review]
LGTM. Two typos only (before this gets fixed by someone else in a separate commit) 😂

File: hal/halx86/apic/tsc.c (hunk 5 above)
Comment (tkreuzer) [reply hal/halx86/apic/tsc.c]: OK, I fixed it, but please stop asking for unrelated changes. Especially if it's fucking typos.

**Pull request merged**